client/src/backend/SnarkManager.ts

```diff
@@ -40,44 +41,46 @@ interface BattleSnarkInput {
   yourCommitment: string;
 }
 
+type SnarkInput = InitSnarkInput | MoveSnarkInput | BattleSnarkInput;
+
 type ZKPTask = {
   taskId: number;
-  input: unknown;
+  input: SnarkInput;
   circuit: string; // path
   zkey: string; // path
 };
 
-type SnarkInput = InitSnarkInput | MoveSnarkInput | BattleSnarkInput;
-
 export class SnarkProverQueue {
-  private taskQueue: FastQueue.queueAsPromised<ZKPTask, SnarkJSProofAndSignals>;
-  private taskCount: number;
+  #taskQueue: FastQueue.queueAsPromised<ZKPTask, SnarkJSProofAndSignals>;
+  #taskCount: number;
+  #snarkjs: Comlink.Remote<{
+    groth16: {
+      fullProve: (input: SnarkInput, circuit: string, zkey: string) => SnarkJSProofAndSignals;
+    } & Comlink.ProxyMarked;
+  }>;
 
   constructor() {
-    this.taskQueue = FastQueue.promise(this, this.execute, 1);
-    this.taskCount = 0;
+    this.#taskQueue = FastQueue.promise(this, this.execute, 1);
+    this.#taskCount = 0;
+    this.#snarkjs = Comlink.wrap(new Worker(new URL('/snarkjs.worker.js', import.meta.url)));
```

Comment:
ah I was hoping this change meant that the worker js can be imported via package managers, guessing that's not possible?

Reply:
Vite has multiple ways to import a worker file, but using `new Worker(new URL('/filename.js', import.meta.url))` is the recommended way.